met-web/src/components/engagement/form/EngagementWidgets/Documents/CreateFolderForm.tsx

```diff
@@ -39,17 +39,17 @@ const CreateFolderForm = () => {
     };
 
     const handleCreateFolder = async () => {
-        if (!widget || validate()) {
-            return;
-        }
-
         try {
+            if (!validate()) {
+                return;
+            }
             setCreatingFolder(true);
-            await postDocument(widget.id, {
-                title: folderName,
-                widget_id: widget.id,
-                type: DOCUMENT_TYPE.FOLDER,
-            });
+            if (widget)
```

Comment:
after the change to every if (!widget || !validate()) didn't work?

Reply:
Typescript gives linting errors for widget without the conditional